onnxscript/rewriter/_rewrite_rule.py

```diff
@@ -257,7 +337,75 @@ def replace_pattern(new_pattern):
         return [replace_pattern(p) for p in self._target_pattern.commute()]
 
 
-class RewriteRuleClassBase(abc.ABC):
+class PatternBase(abc.ABC):
+    """Base class for implementing pattern matching as a class.
+
+    This class encapsulates the pattern definition and condition checking
+    without the replacement functionality.
+
+    Example::
+
+        class TransposePattern(PatternBase):
+            def pattern(cls, op, x, perm):
+                return op.Transpose(x, perm=perm)
+
+            def check(cls, context, x: ir.Value, perm: ir.Attr) -> bool:
+                if perm.is_ref():
+                    return False
+                if perm.type == ir.AttributeType.INTS:
+                    if perm.as_ints() == list(range(len(perm.as_ints()))):
+                        return True
+                return False
+    """
+
+    def __init__(self, name: str | None = None, **kwargs) -> None:
+        self.name = name or self.__class__.__name__
+        # Create and store the Pattern internally
+        self._compiled_pattern = Pattern(self.pattern, self.check, name=self.name, **kwargs)
```

Comment:
This is causing a CI failure. The problem is that we should construct `Pattern(...)` on the right hand side of above assignment only after the constructor of the derived class is complete. Can you initialize this field to None here, and then construct in on demand in the `match` method and assign it to this field (if it is None)?